test_vm/tests/replica_update_test.rs

```diff
@@ -26,14 +28,920 @@ use fvm_shared::sector::{RegisteredSealProof, SectorNumber};
+            .flush()
+            .unwrap();
+    assert_eq!(deadline_state.sectors_snapshot, empty_sectors_array);
+
+    assert!(miner_power(&v, miner_id).is_zero());
+
+    declare_recovery(&v, worker, miner_id, deadline_index, partition_index, sector_number);
+    let (deadline_info, _, v) = advance_to_proving_deadline(v, miner_id, sector_number);
+
+    submit_windowed_post(
+        &v,
+        worker,
+        miner_id,
+        deadline_info,
+        partition_index,
+        PowerPair {
+            raw: StoragePower::from(sector_size as i64),
+            qa: StoragePower::from(sector_size as i64),
+        },
+    );
+
+    assert!(check_sector_active(&v, miner_id, sector_number));
+    assert!(!check_sector_faulty(&v, miner_id, deadline_index, partition_index, sector_number));
+    assert_eq!(miner_power(&v, miner_id).raw, BigInt::from(sector_size as i64));
+}
+
+#[test]
+fn prove_replica_update_multi_dline() {
+    let store = &MemoryBlockstore::new();
+    let policy = Policy::default();
+    let mut v = VM::new_with_singletons(store);
+    let addrs = create_accounts(&v, 1, TokenAmount::from(100_000e18 as i128));
+    let (worker, owner) = (addrs[0], addrs[0]);
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
+    let (maddr, _) = create_miner(
+        &mut v,
+        owner,
+        worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        TokenAmount::from(10_000e18 as i128),
+    );
+
+    v = v.with_epoch(1440); // something offset far away from deadline 0 and 1
+
+    /* Commit enough sectors to pack two partitions */
+    let more_than_one_partition = 2440;
+    let batch_size = 100;
+    let first_sector_number_p1 = 0;
+    let first_sector_number_p2 = seal_proof.window_post_partitions_sector().unwrap();
+    let expiration = v.get_epoch() + policy.max_sector_expiration_extension;
+
+    let new_precommits = precommit_sectors(
+        &mut v,
+        more_than_one_partition,
+        batch_size,
+        worker,
+        maddr,
+        seal_proof,
+        first_sector_number_p1,
+        true,
+        Some(expiration),
+    );
+
+    let precommits = new_precommits;
+    let to_prove = precommits;
+
+    let prove_time = v.get_epoch() + policy.pre_commit_challenge_delay + 1;
+    v = advance_by_deadline_to_epoch(v, maddr, prove_time).0;
+
+    prove_commit_sectors(&mut v, worker, maddr, to_prove, batch_size);
+
+    /* This is a mess, but it just ensures activation of both partitions by posting, cronning and checking */
+
+    // advance to proving period and submit post for first partition
+    let (deadline_info, partition_index, v) =
+        advance_to_proving_deadline(v, maddr, first_sector_number_p1);
+
+    // first partition shouldn't be active until PoSt
+    assert!(!check_sector_active(&v, maddr, deadline_info.index,));
+    submit_windowed_post(&v, worker, maddr, deadline_info, partition_index, PowerPair::zero());
```

Comment:
How is this not failing? All of these sectors should be going from unproven to proven and activating lots of power.

Reply:
Ah I see this is a bit of a bug in the submit_window_post helper, we should be asserting no subinvocations instead of not checking at all in the case of zero power.  Not critical for you to fix here but it would be welcome.

Reply:
Yeah, I actually really hate that -- opened this issue a few days ago, @geoff-vball is on it.